### DISCUSSION TITLE
Mastodon can't just choose to not add markdown anymore.

### DISCUSSION THREAD

**Discussion by Freeplayg** [mastodon/mastodon#18872] [General]
I guess this would make more sense as a discussion.

Mastodon is the biggest Fediverse platform. And with every single other Fediverse platform already supporting basic markdown, new _and_ old... So, so much information gets lost. Bold, italics, etc. have so much meaning and can dramatically change how a sentence is read.

Not even blockquotes are visible on Mastodon, it's all just plain text. How am I supposed to know that it's a quote? You can't.

I love this platform, and don't want these disconnects between platforms. It doesn't matter if you just don't like it. You can't just choose to not add it anymore.

**Comment by wizzwizz4** [comment]
You don't need to be able to type it, so supporting _Markdown_ isn't necessary – but you _do_ need to be able to display posts as-written when they're simple rich text.

Compare the above, with different emphasis:

> You don't _need_ to be able to type it, so supporting Markdown isn't _necessary_ – but you do need to be able to display _posts_ as-written when they're simple rich text.

The meaning's quite different! Most people using social media aren't thinking about formatting incompatibility when sending messages to each other.

**Comment by Andre601** [comment]
Agree on the fact that Mastodon shouldn't just use plain text rendering for the content, as that breaks text entirely.

Here's an example from an instance using glitch-soc, which adds Markdown and HTML formatting support:
![grafik](https://user-images.githubusercontent.com/11576465/201499770-f5367f36-cfad-4c53-8580-4d8b4d5df745.png)

And for comparison is here the same post but viewed with the official Android app:
![grafik](https://user-images.githubusercontent.com/11576465/201499809-a6f6d480-d0c1-4dd6-9693-bbc8946ade4a.png)

Weirdly enough are embedded links supported, but lists aren't. Not even a basic line break is used in that case and instead everything is *squeezed* together in an unreadable mess.

I sadly can't test how other vanilla Mastodon instances render this post (URL is [here](https://blobfox.coffee/@purrbot/109314576557102942). Would appreciate if someone can share the result), but I would assume it's probably the same?

Either way, even if Markdown wouldn't be supported (Let's be honest: It should. It's way to good to not implement), should posts at least look the same across instances and not break appart if one doesn't use the same version and/or extensions/additions.

IF Markdown would ever be added should there be an option to see a preview of the content, so that you can correct wrong formatting before posting.

**Reply by raf64flo** [reply]
Here is the rendering using [Tusky app](https://tusky.app/):

![image](https://user-images.githubusercontent.com/133268/201537226-0967e203-8c7d-49e2-8140-3db8d7153dcb.png)

**Reply by jakeyounglol** [reply]
this is how it looks in ivory for mastodon (iOS)
![ee](https://github.com/user-attachments/assets/4da5d92b-371f-4333-b091-1184658c8cca) and here’s how it looks on my instance website (i have the tangerine ui for mastodon userscript installed in my browser)
![ee](https://github.com/user-attachments/assets/2c239b89-f4af-473e-b10d-005f2b083bea)

**Comment by randytayler** [comment]
I love the idea of adding markdown. I don't know why Twitter never implemented it. Seems like a straightforward feature to accept underscores and double-asterisks (like GitHub does here) - I'd love to understand why it _wouldn't_ be straightforward.

**Comment by Freeplayg** [comment]
I've made an issue for it here btw https://github.com/mastodon/mastodon/issues/18958

Please upvote it!!

**Comment by McPringle** [comment]
Markdown is supported by [Elk](https://elk.zone/), too:

![image](https://github.com/mastodon/mastodon/assets/1254039/458866d4-da34-45a5-91c3-5e565827dc12)

It looks like nearly all other 3rd party clients support Markdown and the Mastodon web client itself is the last one missing this feature.

**Comment by renchap** [comment]
Mastodon supports rendering rich text statuses since 4.2 (or 4.1?)

Authoring rich text statuses is on the roadmap, we looked at using the feature from Glich SOC but is has some issues that we have not been able to solve yet.

**Comment by McPringle** [comment]
> Mastodon supports rendering rich text statuses since 4.2 (or 4.1?)

The original Mastodon web client in v4.2.3 (latest, released two weeks ago) does **not** support Markdown (this issue is about Markdown, not a HTML subset or other formats):

![image](https://github.com/mastodon/mastodon/assets/1254039/f709a225-6400-4c95-8f56-5cb07fc202a0)

Source: https://fosstodon.org/@McPringle/111601401605446538

**Reply by renchap** [reply]
Rich text statuses are in ActivityPub should be exchanged as HTML, and then Mastodon will render them properly.

There are no plans to support rendering text posts as Markdown

**Reply by McPringle** [reply]
I highly appreciate all the work done by all the open source developers. I'm myself an open source developer (mainly Java and Kotlin projects), so I know that it means to use the rare spare time to contribute to OSS projects. So please take my following words not as a complaint, but as my personal opinion with my above background.

My opinion is exactly the same as the OP wrote in the title of this discussion:

> Mastodon can't just choose to not add markdown anymore.

In the last weeks I tested several Mastodon clients for my phone and my desktop PC. The only one I found not supporting the rendering of Markdown is the original Mastodon client. I respect the decisions of all the people managing this open source project. I know they put all their heart and love into this project. But I don't understand the decision not to support Markdown. To me, it **feels** narrow-minded. Markdown is kind of a standard and supported in so many tools all over the internet.

For me, the main reason why I use 3rd party applications on my phone and desktop is the lack of multi-account-support. I have a private account and in addition, I manage two community accounts. The second biggest benefit of using 3rd party apps is the support of Markdown rendering. I use Markdown all day long. I know *a lot of people* who are using Markdown a lot.

I respect your decision, but I also regret it. I hope that you will revise it.

**Reply by ghost** [reply]
Hi @renchap,

When you say:
> Rich text statuses are in ActivityPub should be exchanged as HTML, and then Mastodon will render them properly.

What does that imply for the end user? That I should be able to paste rich text into the client and the formatting will be honored? I've just tried doing that on the Mastodon web UI (using formatted text from the preview here in the comment composer), but it doesn't seem to be the case.

**Reply by renchap** [reply]
Mastodon does not allow writing rich text at the moment. It only supports displaying it for statuses that have been created with other software and that have been federated.

**Reply by ghost** [reply]
Thanks for the clarification 👍🏻 

Edited to remove my question, just saw you answered it in a post above.

**Comment by seaglade** [comment]
Arguably better would be supporting a subset of HTML (removing `head`, `body`, `style` and `script` tags among others, for obvious reasons), and a rich text option when authoring a post that generates HTML for the actual ActivityPub content. It wouldn't be too hard to also support converting Markdown from the user input to HTML for the actual data.

**Reply by seaglade** [reply]
TL;DR: Instead of supporting *rendering* Markdown, Mastodon could support converting Markdown to HTML when a post is authored, and then rendering that HTML.

**Comment by ghost** [comment]
If mastodon had markdown or bb codes support I would disable the ability of users to upload images & videos and force them to use image hosting websites. 

Forcing them to use image hosting websites would allow me to run my mastodon instance without worrying about HD space. 

In fact my instance is just me for now because I can't deal with the HD space requirements. 

Markdown or BB Codes is needed badly.
